html5ever/src/serialize/mod.rs

```diff
@@ -94,6 +107,30 @@ impl<Wr: Write> HtmlSerializer<Wr> {
 }
 
 impl<Wr: Write> Serializer for HtmlSerializer<Wr> {
+    fn start_without_write(&mut self, name: QualName) {
```

Comment:
I'm not sure I understand why this is an improvement over my suggested solution in https://github.com/servo/servo/issues/14975#issuecomment-307263635 . Am I right that the code that uses the serializer would choose to call start_without_write or start_elem with the parent node, based on whether ChildrenOnly is selected? This seems like it makes the serializer interface more complicated than necessary to me.